pkg/cloudprovider/fake/capacity.go

```diff
@@ -57,3 +57,19 @@ func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Constr
 func (c *Capacity) Delete(ctx context.Context, nodes []*v1.Node) error {
 	return nil
 }
+
+func (c *Capacity) GetZones(ctx context.Context) ([]string, error) {
+	return []string{"test-zone"}, nil
+}
+
+func (c *Capacity) GetInstanceTypes(ctx context.Context) ([]string, error) {
+	return []string{"test-instance-type"}, nil
+}
+
+func (c *Capacity) GetArchitectures(ctx context.Context) ([]string, error) {
+	return []string{"test-architecture"}, nil
+}
+
+func (c *Capacity) GetOperatingSystems(ctx context.Context) ([]string, error) {
+	return []string{"test-operating-system"}, nil
+}
```

Comment:
Maybe -- ish. Let me see if there's a clean way to do this. for each testing was getting a bit nasty since I need to re-randomize the resource names every time.

Reply:
Compromise. I'll support multiple (in the fake cloud provider), but not test all of them exhaustively.